cocos/game/splash-screen.ts

```diff
@@ -66,181 +66,171 @@ interface SplashLogo {
     image?: string;
     base64?: string;
 }
-const _dsInfo = new DescriptorSetInfo(null!);
+
 export class SplashScreen {
-    private settings!: ISplashSetting;
-    private _curTime = 0;
-
-    private device!: Device;
-    private swapchain!: Swapchain;
-    private shader!: Shader;
-    private sampler!: Sampler;
-    private cmdBuff!: CommandBuffer;
-    private quadAssmebler!: InputAssembler;
-    private vertexBuffers!: Buffer;
-    private indicesBuffers!: Buffer;
-    private renderArea!: Rect;
-    private clearColors!: Color[];
-    private projection!: Mat4;
-    private isMobile = false;
-
-    private bgMat!: Material;
-    private bgImage!: HTMLImageElement;
-    private bgTexture!: Texture;
-
-    private logoMat!: Material;
-    private logoImage!: HTMLImageElement;
-    private logoTexture!: Texture;
-
-    private watermarkMat!: Material;
-    private watermarkTexture!: Texture;
+    private settings$!: ISplashSetting;
+    private _curTime$ = 0;
+
+    private device$!: Device;
+    private swapchain$!: Swapchain;
+    private shader$!: Shader;
+    private sampler$!: Sampler;
+    private cmdBuff$!: CommandBuffer;
+    private quadAssmebler$!: InputAssembler;
+    private vertexBuffers$!: Buffer;
+    private indicesBuffers$!: Buffer;
+    private renderArea$!: Rect;
+    private clearColors$!: Color[];
+    private projection$!: Mat4;
+    private isMobile$ = false;
+
+    private bgMat$!: Material;
+    private bgImage$!: HTMLImageElement;
+    private bgTexture$!: Texture;
+
+    private logoMat$!: Material;
+    private logoImage$!: HTMLImageElement;
+    private logoTexture$!: Texture;
+
+    private watermarkMat$!: Material;
+    private watermarkTexture$!: Texture;
 
     // layout
-    private bgWidth = 1920;
-    private bgHeight = 1080;
-    private logoWidthTemp = 140;
-    private logoHeightTemp = 200;
-    private logoWidth = 0;
-    private logoHeight = 0;
-    private logoXTrans = 1 / 2;// Percent
-    private logoYTrans = 1 / 6 + 2.5 / 6;// Percent
-
-    private textSize = 24; // font size
-    private textHeight = 24; // line height
-    private textXTrans = 1 / 2;// Percent
-    private textYExtraTrans = 32;// px
-    private textExpandSize = 4;// px
-
-    private scaleSize = 1;
+    private bgWidth$ = 1920;
+    private bgHeight$ = 1080;
+    private logoWidthTemp$ = 140;
+    private logoHeightTemp$ = 200;
+    private logoWidth$ = 0;
+    private logoHeight$ = 0;
+    private logoXTrans$ = 1 / 2;// Percent
+    private logoYTrans$ = 1 / 6 + 2.5 / 6;// Percent
+
+    private textSize$ = 24; // font size
+    private textHeight$ = 24; // line height
+    private textXTrans$ = 1 / 2;// Percent
+    private textYExtraTrans$ = 32;// px
+    private textExpandSize$ = 4;// px
+
+    private scaleSize$ = 1;
 
     public get isFinished (): boolean {
-        if (WECHAT) {
-            return true;
-        }
```

Comment:
Why remove it?

Reply:
It's controlled by `rollup treeshaking` functionality to remove the code. You could refer to the changes in game.ts.
We just need to add `if (WECHAT)` code in where we invoke `SplashScreen` rather than in every functions in `splash-screen.ts`